docs: add transaction management how-to and related new terms (#416)
nervosnetwork/docs.nervos.org: website/docs/how-tos/how-to-manage-txs.mdx

---

import useBaseUrl from "@docusaurus/useBaseUrl";
import Tooltip from "@components/Tooltip";

# How to Manage Transactions

Contributor: [@doitian](https://github.com/doitian)

Managing UTXO transactions can be challenging because they may be stuck in or even dropped from the CKB node transactions pool when the fee is too low.

This guideline is for any transaction generator looking to trace the transaction status before it is finally confirmed on the chain.

## Transaction States

The transaction lifecycle starts after creation and ends when it is finally confirmed or conflicts with a confirmed transaction. During this lifecycle, the transaction must be in one of the following states:

- Pending
- Confirming
- Confirmed
- Conflicting
- Conflictive
- Reverted
- Abandoned

The list excludes <Tooltip>orphan transaction</Tooltip> that depends on unknown Cells, as transaction generators typically ignore them.

Different states require different strategies as described in the following sections.

<img
  alt="Transition Between Different Transaction States"
  src={useBaseUrl("img/how-tos/transaction-states.png")}
  width="688"
/>

## Pending Transaction

> The newly created transaction enters the **Pending** state.

Reviewer: ```suggestion
A newly created transaction starts in the **Pending** state.
```

The generator must store pending transactions locally and regularly send them to CKB nodes. It's essential because CKB nodes may drop transactions from their pools.

> It is not recommended for the generator to use output Cells of pending transactions unless the transaction is stuck and requires a <Tooltip>fee bumping</Tooltip>. If chaining pending transactions is necessary, limit the chain’s length.

Reviewer: ```suggestion
It is not recommended for the generator to use output Cells of pending transactions unless the transaction is stuck and requires a <Tooltip>fee bumping</Tooltip>. If chaining pending transactions is unavoidable, limit the chain’s length.
```

> Once the transaction appears in a block on the canonical chain, it turns to **Confirming**.

Reviewer: ```suggestion
The transaction turns to the **Confirming** state once it appears in a block on the canonical chain.
```
